### PR TITLE
bitnami/kafka add minBrokerId variable

### LINKED ISSUE #4626: bitnami/kafka no possibility to set an arbitrary KAFKA_CFG_MIN_BROKER_ID
**Which chart**:
bitnami/kafka

**Is your feature request related to a problem? Please describe.**
In current kafka chart it is not possible to set an arbitrary `KAFKA_CFG_MIN_BROKER_ID` for a stateful set with multiple pods. The `KAFKA_CFG_BROKER_ID` value is hardcoded via `/scripts/setup.sh` to the ID of the pod in the stateful set. This makes impossible to provision a stretched cluster where two separate helm releases form a single kafka cluster, because BROKER_IDs in the stateful set always start from zero. Such stretched clusters are absolutely needed in zero-downtime migration scenarios.
 
**Describe the solution you'd like**
A possibility to set `KAFKA_CFG_MIN_BROKER_ID` variable in the chart. Brokers derive their IDs from the following formula: `KAFKA_CFG_BROKER_ID = KAFKA_CFG_MIN_BROKER_ID + POD_ID`, e.g. if KAFKA_CFG_MIN_BROKER_ID=10, brokers get ID=10,11,12 etc.
If `KAFKA_CFG_MIN_BROKER_ID`variable is set in the chart `setup.sh` script [here](https://github.com/bitnami/charts/blob/master/bitnami/kafka/templates/scripts-configmap.yaml) takes it into account while setting `KAFKA_CFG_BROKER_ID`.

**Describe alternatives you've considered**
A current alternative is to deactivate `/scripts/setup.sh` in the chart completely, set `KAFKA_CFG_BROKER_ID` in the chart manually and provision a separate stateful set per broker, what leads to huge management overhead.

### PR DESCRIPTION
**Pull request by antonmatsiuk**
**Description of the change**

**Benefits**
As described [here](https://github.com/bitnami/charts/issues/4626#issuecomment-740569976) The PR adds `minBrokerId` variable to kafka chart to support stretched clusters in zero-downtime migration scenarios. This allows to set an arbitrary minBrokerId and consequently an arbitrary minimal `broker.id` of kafka brokers in the cluster. This allows to combine brokers from a helm release with other brokers or releases into a single kafka cluster. 

**Possible drawbacks**
no

**Applicable issues**
no 

  - fixes #4626

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/kafka/templates/scripts-configmap.yaml [hunk 1]
```diff
@@ -72,7 +72,7 @@ data:
     #!/bin/bash
 
     ID="${MY_POD_NAME#"{{ $fullname }}-"}"
-    export KAFKA_CFG_BROKER_ID="$ID"
+    export KAFKA_CFG_BROKER_ID="$((ID + {{ .Values.minBrokerId }}))"
```
Comment (juan131) [inline bitnami/kafka/templates/scripts-configmap.yaml:75]: Possible workaround to help upgrading charts when a different ID is set in the existing data:

```console
if [[ -f "/bitnami/kafka/data/meta.properties" ]]; then
    export KAFKA_CFG_BROKER_ID="$(grep "broker.id" /bitnami/kafka/data/meta.properties | awk -F '=' '{print $2}')"
else
    ID="${MY_POD_NAME#"{{ $fullname }}-"}"
    export KAFKA_CFG_BROKER_ID="$ID"
fi
```

File: bitnami/kafka/values-production.yaml [hunk 2]
```diff
@@ -348,6 +348,10 @@ interBrokerListenerName: INTERNAL
 ##
 replicaCount: 3
 
+## Minimal broker.id value
+##
+minBrokerId: 0
```
Comment (juan131) [inline bitnami/kafka/values-production.yaml:353]: Small suggestion to explain how it works
```suggestion
## Minimal broker.id value
## Brokers increment their ID starting at this minimal value.
## E.g., with `minBrokerId=100` and 3 nodes, IDs will be 100, 101, 102 for brokers 0, 1, and 2, respectively.
##
```

File: bitnami/kafka/values.yaml [hunk 3]
```diff
@@ -348,6 +348,10 @@ interBrokerListenerName: INTERNAL
 ##
 replicaCount: 1
 
+## Minimal broker.id value
+##
+minBrokerId: 0
```
Comment (juan131) [inline bitnami/kafka/values.yaml:353]: Small suggestion to explain how it works
```suggestion
## Minimal broker.id value
## Brokers increment their ID starting at this minimal value.
## E.g., with `minBrokerId=100` and 3 nodes, IDs will be 100, 101, 102 for brokers 0, 1, and 2, respectively.
##
```

**Review by juan131 — COMMENTED** [review]
Hi @antonmatsiuk 

Thanks so much for this great contribution!! It can help many users trying to upgrade

Take a look to my comments

File: bitnami/kafka/templates/scripts-configmap.yaml (hunk 1 above)
Comment (antonmatsiuk) [reply bitnami/kafka/templates/scripts-configmap.yaml:75]: @juan131 but how can one populate `/bitnami/kafka/data/meta.properties`? or does it get parsed and populated from `/bitnami/kafka/config/server.properties` which in turn can be configured via`config:` in the chart?

Comment (juan131) [reply bitnami/kafka/templates/scripts-configmap.yaml:75]: Id' try to explain myself. Let's say you install the chart with 3 brokers:

```console
$ helm install my-release bitnami/kafka --set replicaCount=3 --set minBrokerId=100
``` 

We can see each broker has its own PVC mounted at `/bitnami/kafka` to persist the Kafka data.

```console
$ kg pvc
NAME                     STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-kafka-0             Bound    pvc-14dc5cb4-83d7-41f4-b584-e4e0924c915e   8Gi        RWO            standard       69s
data-kafka-1             Bound    pvc-1c2539bf-0e02-4107-9879-8cbbe9eb04b0   8Gi        RWO            standard       69s
data-kafka-2             Bound    pvc-5b8dfb2b-8193-4721-ac82-1de9e360d962   8Gi        RWO            standard       69s
```

And... If we inspect the `meta.properties` the brokers:

```console
$ kubectl exec kafka-0 -- grep "broker.id" /bitnami/kafka/data/meta.properties
broker.id=100
$ kubectl exec kafka-1 -- grep "broker.id" /bitnami/kafka/data/meta.properties
broker.id=101
$ kubectl exec kafka-2 -- grep "broker.id" /bitnami/kafka/data/meta.properties
broker.id=102
```

Now.. let's say that you upgrade the chart (but you forget to set **minBrokerId**):

```console
$ helm upgrade kafka bitnami/kafka --set replicaCount=3
```

You'll find the error below:

```
$ kubectl logs kafka-2
...
kafka.common.InconsistentBrokerIdException: Configured broker.id 2 doesn't match stored broker.id 102 in meta.properties
```

With my proposal.. You avoid that error by ensuring you respect the broker id persisted in the volumes.

Comment (antonmatsiuk) [reply bitnami/kafka/templates/scripts-configmap.yaml]: @juan131 agree, just pushed the necessary changes.

File: bitnami/kafka/values-production.yaml [hunk 4]
```diff
@@ -348,6 +348,11 @@ interBrokerListenerName: INTERNAL
 ##
 replicaCount: 3
 
+## Minimal broker.id value
+## Brokers increment their ID starting at this minimal value.
+## E.g., with `minBrokerId=100` and 3 nodes, IDs will be 100, 101, 102 for brokers 0, 1, and 2, respectively.
+##
+
```
Comment (juan131) [inline bitnami/kafka/values-production.yaml:355]: `minBrokerId` should be set to `0` by default:
```suggestion
minBrokerId: 0
```

File: bitnami/kafka/values.yaml [hunk 5]
```diff
@@ -348,6 +348,11 @@ interBrokerListenerName: INTERNAL
 ##
 replicaCount: 1
 
+## Minimal broker.id value
+## Brokers increment their ID starting at this minimal value.
+## E.g., with `minBrokerId=100` and 3 nodes, IDs will be 100, 101, 102 for brokers 0, 1, and 2, respectively.
+##
+
```
Comment (juan131) [inline bitnami/kafka/values.yaml:355]: `minBrokerId` should be set to `0` by default:
```suggestion
minBrokerId: 0
```

File: bitnami/kafka/Chart.yaml [hunk 6]
```diff
@@ -29,4 +29,4 @@ name: kafka
 sources:
   - https://github.com/bitnami/bitnami-docker-kafka
   - https://kafka.apache.org/
-version: 12.3.1
+version: 12.3.2
```
Comment (juan131) [inline bitnami/kafka/Chart.yaml:32]: Please bump minor version instead, since we're adding a new parameter.

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/kafka/Chart.yaml (hunk 6 above)
Comment (antonmatsiuk) [reply bitnami/kafka/Chart.yaml]: minor version added with the latest commit

**Review by juan131 — APPROVED** [review]
Thanks for applying the sugggestions! LGTM

**Pull request merged**